app/models/user.rb

```diff
@@ -11,4 +11,8 @@ def assign_from_auth(auth)
     self.name = auth_attrs.name
     self.uid = auth.uid
   end
+
+  def with_asserted_attributes
+    UserWithAssertedAttributes.new(self)
+  end
```

Comment:
This change looks great, except for the addition of this method. I don't think this function is the responsibility of the User class. I'm not sure this helper method is needed since the instantiation is only called once. We want the two objects to be separate entities. 

The other advantage of having a separate class to handle the asserted attributes is it allows us to deal with immutable objects. Given the same input to the `UserWithAssertedAttributes` class, we always get the same output, in the form of a new object that is guaranteed to never change afterwards. Before, we were starting with a User object, and getting that same object back with a different state, and with the possibility that it might change again.